helm/adelphi/templates/gemini-results.yaml

```diff
@@ -0,0 +1,36 @@
+apiVersion: argoproj.io/v1alpha1
+kind: WorkflowTemplate
+metadata:
+  name: gemini-results
+  namespace: {{ .Values.namespace }}
+  annotations:
+    "helm.sh/hook": pre-install
+spec:
+  templates:
+  - name: collect-results
+    container:
+      image: alpine
+      command: ["sh", "-c", "-e"]
+      args:
+        - apk add jq;
+          export TIMESTAMP=`date +%s`;
+          push()
+          {
+            METRIC=$1
+            shift;
+            cat /results/results.json |
+            jq -r ".result.$METRIC" |
+            xargs -i echo "gemini.$METRIC {} $TIMESTAMP" |
+            nc graphite-service 9109;
+          };
+          push "write_ops";
+          push "write_errors";
+          push "read_ops";
+          push "read_errors";
```

Comment:
This feels like the kind of thing that would be better expressed via a script that could just be run somewhere rather than an inline collection of shell commands like this

Reply:
@absurdfarce I like shell commands in workflow templates, it makes it pretty easy to track where/when it gets executed. Between that and a separate script, I don't see much difference.
The ideal solution would be patching Gemini to send the metrics directly as it generates them, so we'd actually have a time-series instead of just a single data point at the end. I'm planning to tackle that along with other Gemini work we have planned.
In the meantime, this should be fine.

Reply:
I was debating something similar with myself 😄  recently w.r.t. to GH Actions, ultimately I came to preferring the commands inline in the action, similar to what you've got here.  What makes the decision easier I think is would the contents of that script be used by anything but this?  If it would be, then a separate script makes sense, if not, I like the ease of being able to understand what will happen directly from the template.